pkg/data/data_store.go

```diff
@@ -23,7 +23,7 @@ type Store struct {
 }
 
 // NewStore creates a new Store instance
-func NewStore(database *mgo.Database, logger logging.Logger) *Store {
+func NewStore(database *mgo.Database, logger logging.Logger) IDataStore {
```

Comment:
I made this change because the mock will only work if DataStore returns its interface instead of its struct. But anyway I changed the name of the interface, just as you indicated.